spec/src/main/asciidoc/core/events.asciidoc

```diff
@@ -296,15 +296,18 @@ An event parameter may have multiple qualifiers.
 public void afterDocumentUpdatedByAdmin(@Observes @Updated @ByAdmin Document doc) { ... }
 ----
 
-Then this observer method will only be notified if all the observed event qualifiers are specified when the event is fired:
+Then this observer method will be notified if the set of observer qualifiers is a subset of the fired event's qualifiers:
 
 [source, java]
 ----
 @Inject Event<Document> documentEvent;
 ...
-documentEvent.select(new UpdatedQualifier(), new ByAdminQualifier()).fire(document);
+documentEvent.select(new UpdatedQualifier(), new ByAdminQualifier(), new ClarificationQualifier()).fire(document);
 ----
 
+In the above example the event is fired with `@ByAdmin`, `@Updated` and `@Clarification` qualifiers while the observer qualifiers are `@Updated` and `@ByAdmin`.
+Observer qualifiers therefore form a subset of event qualifiers and the observer will be notified.
```

Comment:
I think @johnament mention that `@Any` is useless regarding event. You don't need to use it for injection if you want to select an event of a given qualifier later.
Regarding empty set I agree, the problem is that it's working differently than typesafe resolution but look very similar to programmatic lookup. We should find a way to clarify this ambiguity.